nest/04.finest.md

```diff
@@ -0,0 +1,191 @@
+├─ @Group1
+|  ├─ requirement1
+|  |  └─ fullfiled by package1  
+|  └─ requirement2
+|     └─ fulfilled by package2
+└─ ...
+```
+
+## 4.2 finest commands
+### 4.2.1 Managing packages through requirements
+#### Add a requirement
+The command `finest requirement add <requirement> [--parent <group>]` adds a requirement to a group.
+By default, this group is `@root`.
+With the `--parent` flag, you can speficy a parent group.
+
+When adding a requirement, `finest` makes a copy of your current dependency graph (if a copy wasn't already available), adds that requirement to its parent and fulfills the dependencies of the package recursively.
+
+Its output looks like this:
+
+```
+$ finest requirement add linux
+Adding requirement stable::kernel/linux#* to group @root...
+```
+
+#### Remove a requirement
+`finest requirement remove <requirement> [--parent <group>]` removes a requirement from a group.
+The group is, by default, `@root`, but you can specify a parent group with the flag `--parent`.
+
+When removing a requirement, `finest` makes a copy of your current dependency graph (if a copy wasn't already made), removes that requirement from its parent and removes its depedencies recursively if possible.
+
+Its output is similar to this:
+
+```
+$ finest requirement remove linux --parent @work
+Removing requirement stable::kernel/linux#* to group @work...
+```
+
+#### Common issues
+* `error: no package found for requirement <req>`
+
+This error means that `finest` couldn't find any package to satisfy this requirement.
+What probably went wrong, is that the syntax of the requirement is incorrect.
+You can check on [Raven-OS website](https://stable.raven-os.org/) to check the repository and the category of this requirement.
+
+### 4.2.2 Updating the system by merging the depedency graph
+#### Merging the dependency graph
+`finest merge` merges the copy of your dependency graph, on which `finest` applied the transactions, to your current dependency graph.
+These two graphs are compared to extract a list of transactions that `finest` will perform, in order to go from one graph to another.
+And so, your dependency graph will finally apply the transactions you meant to do.
+
+Example:
+```
+2 pending transactions:
+
+ install stable::kernel/linux#4.14.96
+ remove stable::sys-libs/glibc#2.0.1
+
+Would you like to apply these transactions? [Yes/no] Y
+Downloading stable::kernel/linux#4.14.96...
+Successfully installed stable::kernel/linux#4.14.96
+Removing stable::sys-libs/glibc#2.0.1...
+Successfully uninstalled stable::sys-libs/glibc#2.0.1
+```
+
+> Important
+>
+> Never forget to run `finest merge` after you're done with your requirements, or they won't be applied on the system
+
+### 4.2.3 Managing groups
+#### Create a group
+Creating a group is possible via the command `finest group create <name> [--parent <group>]`.
+It's very important to keep in mind that a group name always starts with a `@` character.
```

Comment:
Is it important? It's possible to not respect this rule? Or won't finest just throw an error?

Reply:
yep it just throws an error, I don't know why I overstated it